### PR TITLE
Added Cleaner tool

### PR DESCRIPTION
**Pull request by RandomModderJDK**
Added Cleaner tool, which helps keep the downloads and bookmarks organized by removing watched videos.
First step to implement issue [#1125](https://github.com/SkyTubeTeam/SkyTube/issues/1125)

### CODE REVIEW

**Comment by gzsombor** [comment]
Strange, that it can't build the app, even though JitPack has the necessary file : https://jitpack.io/com/github/SkyTubeTeam/components/okhttp-client/v0.0.5/okhttp-client-v0.0.5.pom and it is possible to download as of now.

**Comment by RandomModderJDK** [comment]
I think the problem is that jitpack shows skytube.components.components as the parent project and not skytube.components

**Comment by gzsombor** [comment]
Can you rebase your changes, I hope I could fix the build.

**Comment by RandomModderJDK** [comment]
Done. Works fine apperantly

File: app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     * @return
+     */
+    private Single<Pair<Integer,Long>> cleanWatchedDownloads(@NonNull Context context) {
+        return Single.fromCallable(() -> {
+                    // Stats
+                    AtomicInteger removedVideosCnt = new AtomicInteger();
+                    AtomicLong removedB = new AtomicLong();
+                    // Maybe rewrite this with reactivex, so that it's parallel
+                    DownloadedVideosDb.getVideoDownloadsDb().getDownloadedVideos().stream()
+                            .filter((video) -> DownloadedVideosDb.getVideoDownloadsDb().isVideoDownloaded(video.getVideoId()).blockingGet()) // Filter only downloaded videos
```
Comment (gzsombor) [inline app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:82]: I'm not sure, I understand this. DownloadedVideosDb.getVideoDownloadsDb().getDownloadedVideos() returns a list of downloaded videos, why do you call the 'isVideoDownloaded' in the next line?

File: app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                            .filter((video) -> PlaybackStatusDb.getPlaybackStatusDb().getVideoWatchedStatus(video.getId()).isFullyWatched()) // Filter only watched videos
+                            .forEach((video) -> {
+                                DownloadedVideosDb.Status status = DownloadedVideosDb.getVideoDownloadsDb().getDownloadedFileStatus(context, video.getVideoId()).blockingGet();
```
Comment (gzsombor) [inline app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:85]: And this also seems to do the same check...

Comment (gzsombor) [reply app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:85]: I see, you used the status later, to get the file sizes... Hmm, maybe we could have a variant of DownloadedVideosDb.getVideoDownloadsDb().getDownloadedVideos() which could return DownloadedVideosDb.Status directly - and we need to add the VideoId to  DownloadedVideosDb.Status as well

File: app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+     * @param context The Context for showing Toasts
+     */
+    private Single<Pair<Integer,Long>> cleanWatchedBookmarks(@NonNull Context context) {
+        return Single.fromCallable(() -> {
+                    // Stats
+                    AtomicInteger removedVideosCnt = new AtomicInteger();
+                    AtomicLong removedB = new AtomicLong();
+                    // Maybe rewrite this with reactivex, so that it's parallel
+                    BookmarksDb.getBookmarksDb().getBookmarkedVideos(BookmarksDb.getBookmarksDb().getTotalBookmarkCount().blockingGet(), null).first.stream()
+                            .filter((video) -> BookmarksDb.getBookmarksDb().isBookmarked(video.getId())) // Filter only bookmarked videos
```
Comment (gzsombor) [inline app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:114]: Here, similar things happens, getBookmarkedVideos returns a list of bookmarked videos, no need for filtering it again.

File: app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java (hunk 1 above)
Comment (RandomModderJDK) [reply app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:82]: Well, you are right, thats unnecessary

File: app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java (hunk 3 above)
Comment (RandomModderJDK) [reply app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:114]: Jup

File: app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java (hunk 2 above)
Comment (RandomModderJDK) [reply app/src/main/java/free/rm/skytube/gui/businessobjects/CleanerDialog.java:84]: should DownloadedVideosDb.status not extend the YoutubeVideo class? Wouldn't that be an option?

**Comment by gzsombor** [comment]
Could you please rebase your changes on master? It seems to have conflicts with that - which is definitely surprising thing...

**Comment by RandomModderJDK** [comment]
Okay, strange. Rebased

**Comment by gzsombor** [comment]
It seems, it's still contains conflicts, so github complains that it can't be rebased.
I can rebase it locally, and open a new pull request, if you don't mind.

**Comment by RandomModderJDK** [comment]
Yeah do that, I mean it states here that it doesnt have any conflicts but okay. Don't know whats wrong

**Comment by gzsombor** [comment]
Thanks for the patch, I will try to create a new release in the near future!

**Pull request closed without merging**